src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -2038,6 +2038,27 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
     // - Also converts \n line-endings to \r line-endings
     void TermControl::_SendPastedTextToConnection(const std::wstring& wstr)
     {
+        // For security reasons, control characters should be filtered.
+        // Here ASCII control characters will be removed, except HT(0x09), LF(0x0a), CR(0x0d) and DEL(0x7F).
+        std::wstring sanitized{ wstr };
+        sanitized.erase(std::remove_if(sanitized.begin(), sanitized.end(), [](wchar_t c) {
+            if (c >= L'\x20' && c <= L'\x7f')
+            {
+                // Printable ASCII characters + DEL.
+                return false;
+            }
+
+            if (c > L'\x7f')
+            {
+                // Not a control character.
+                return false;
+            }
+
+            return c >= L'\x00' && c <= L'\x08' ||
+                   c >= L'\x0b' && c <= L'\x0c' ||
+                   c >= L'\x0e' && c <= L'\x1f';
+        }));
+
         // Some notes on this implementation:

```

Comment:
i'd rather we not scan the string multiple times. can we consolidate this with the loop below?